machinelearning/llama.cpp.md

```diff
@@ -0,0 +1,73 @@
+# Large Language Model (LLM) inference on Graviton CPUs with llama.cpp
+
+**Introduction**
+
+The main goal of [llama.cpp](https://github.com/ggerganov/llama.cpp) is to enable LLM inference with minimal setup and state-of-the-art performance on a wide variety of hardware. It's a plain C/C++ implementation without any dependencies. It supports quantized general matrix multiply and add (GEMM) kernels for faster inference and reduced memory use. The quantized GEMM kernels are optimized for AWS Graviton processors using Arm Neon and SVE based matrix multiply and add (MMLA) instructions. This document covers how to build and run llama.cpp efficiently for LLM inference on AWS Graviton based Amazon EC2 instances.
+
+# How to use llama.cpp on Graviton CPUs
+
+Building from sources is the recommended way to use llama.cpp on Graviton CPUs, and for other hardware platforms too. This section provides the instructions on how to build it from sources.
```

Comment:
Do they not vend an arm64 binary? Is there a specific reason we're recommending building from source? Would include those reasons here.

Reply:
there are no binaries for any other platform too. build from sources is the only option. llama.cpp is a lightweight framework and it takes only couple of minutes to compile from sources.
btw, thanks for verifying the steps.